From ecefb0cb7986ecc83d1e310571e4f5ff90dd2915 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Wed, 21 Sep 2011 09:46:30 +0200
Subject: [PATCH 079/154] pvrvideosink: call pfnWSEGL_DeleteDrawable when
 exiting

Avoids leak and side steps a panic in some kernel cleanup code path.
---
 sys/pvr2d/gstpvrvideosink.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index cf432d8..f144ab5 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -419,9 +419,13 @@ done:
 static void
 gst_pvrvideosink_destroy_drawable (GstPVRVideoSink * pvrvideosink)
 {
-  if (pvrvideosink->dcontext != NULL)
+  if (pvrvideosink->dcontext != NULL) {
+    pvrvideosink->dcontext->wsegl_table->
+        pfnWSEGL_DeleteDrawable (pvrvideosink->dcontext->drawable_handle);
+
     pvrvideosink->dcontext->wsegl_table->
         pfnWSEGL_CloseDisplay (pvrvideosink->dcontext->display_handle);
+  }
 }
 
 /* We are called with the x_lock taken */
-- 
1.7.5.4

